src/arco_era5/update_co.py

```diff
@@ -0,0 +1,177 @@
+    'tcol': ['tclw', 'tciw', 'tcw', 'tcwv', 'tcrw', 'tcsw'],
+    'soil_depthBelowLandLayer_istl1': ['istl1'],
+    'soil_depthBelowLandLayer_istl2': ['istl2'],
+    'soil_depthBelowLandLayer_istl3': ['istl3'],
+    'soil_depthBelowLandLayer_istl4': ['istl4'],
+    'soil_depthBelowLandLayer_stl1': ['stl1'],
+    'soil_depthBelowLandLayer_stl2': ['stl2'],
+    'soil_depthBelowLandLayer_stl3': ['stl3'],
+    'soil_depthBelowLandLayer_stl4': ['stl4'],
+    'soil_depthBelowLandLayer_swvl1': ['swvl1'],
+    'soil_depthBelowLandLayer_swvl2': ['swvl2'],
+    'soil_depthBelowLandLayer_swvl3': ['swvl3'],
+    'soil_depthBelowLandLayer_swvl4': ['swvl4'],
+    'soil_surface_tsn': ['tsn'],
+    'rad': ['ssrd', 'strd', 'str', 'ttr', 'gwd'],  # single-level-forecast
+    'pcp_surface_cp': ['cp'],
+    'pcp_surface_crr': ['crr'],
+    'pcp_surface_csf': ['csf'],
+    'pcp_surface_csfr': ['csfr'],
+    'pcp_surface_es': ['es'],
+    'pcp_surface_lsf': ['lsf'],
+    'pcp_surface_lsp': ['lsp'],
+    'pcp_surface_lspf': ['lspf'],
+    'pcp_surface_lsrr': ['lsrr'],
+    'pcp_surface_lssfr': ['lssfr'],
+    'pcp_surface_ptype': ['ptype'],
+    'pcp_surface_rsn': ['rsn'],
+    'pcp_surface_sd': ['sd'],
+    'pcp_surface_sf': ['sf'],
+    'pcp_surface_smlt': ['smlt'],
+    'pcp_surface_tp': ['tp']
+}
+
+
+def convert_to_date(date_str: str, format: str = '%Y-%m-%d') -> datetime.datetime:
+    return datetime.datetime.strptime(date_str, format)
+
+
+def copy(src: str, dst: str) -> None:
+    cmd = 'gcloud alpha storage cp'
+    try:
+        subprocess.run(cmd.split() + [src, dst], check=True, capture_output=True,
+                       text=True, input="n/n")
+        return
+    except subprocess.CalledProcessError as e:
+        msg = f"Failed to copy file {src!r} to {dst!r} Error {e}"
+        logger.error(msg)
+
+
+@contextmanager
+def opener(fname: str) -> t.Any:
+    _, suffix = os.path.splitext(fname)
+    with tempfile.NamedTemporaryFile(suffix=suffix) as ntf:
+        tmp_name = ntf.name
+        logger.info(f"Copying '{fname}' to local file '{tmp_name}'")
+        copy(fname, tmp_name)
+        yield tmp_name
+
+
+def generate_input_paths(start: str, end: str, root_path: str, chunks: t.List[str], is_single_level: bool = False):
+    input_paths = []
+    for time, chunk in product(pd.date_range(start, end, freq="MS" if is_single_level else "D"), chunks):
+        if is_single_level:
+            url = f"{root_path}/{SINGLE_LEVEL_SUBDIR_TEMPLATE.format(year=time.year, month=time.month, day=time.day, chunk=chunk)}"
+        else:
+            url = f"{root_path}/{MODELLEVEL_SUBDIR_TEMPLATE.format(year=time.year, month=time.month, day=time.day, chunk=chunk)}"
+        
+        if '_' in chunk:
+            chunk_, level, var = chunk.split('_')
+            url = url.replace(chunk, chunk_)
+            url = f"{url}_{level}_{var}.grib"
+        input_paths.append(url)
+
+    return input_paths
+
+
+@dataclass
+class GenerateOffset(beam.PTransform):
+
```

Comment:
nit: `docstrings` for better understanding of transforms.

Reply:
Added at relevant places.